Sources/FoundationEssentials/Subprocess/Platforms/Subprocess+Darwin.swift

```diff
@@ -0,0 +1,232 @@
+        result = posix_spawn_file_actions_adddup2(&fileActions, output.getWriteFileDescriptor().rawValue, 1)
+        guard result == 0 else {
+            try self.cleanupAll(input: input, output: output, error: error)
+            throw POSIXError(.init(rawValue: result) ?? .ENODEV)
+        }
+        if let outputRead = output.getReadFileDescriptor() {
+            // Close parent side
+            result = posix_spawn_file_actions_addclose(&fileActions, outputRead.rawValue)
+            guard result == 0 else {
+                try self.cleanupAll(input: input, output: output, error: error)
+                throw POSIXError(.init(rawValue: result) ?? .ENODEV)
+            }
+        }
+        result = posix_spawn_file_actions_adddup2(&fileActions, error.getWriteFileDescriptor().rawValue, 2)
+        guard result == 0 else {
+            try self.cleanupAll(input: input, output: output, error: error)
+            throw POSIXError(.init(rawValue: result) ?? .ENODEV)
+        }
+        if let errorRead = error.getReadFileDescriptor() {
+            // Close parent side
+            result = posix_spawn_file_actions_addclose(&fileActions, errorRead.rawValue)
+            guard result == 0 else {
+                try self.cleanupAll(input: input, output: output, error: error)
+                throw POSIXError(.init(rawValue: result) ?? .ENODEV)
+            }
+        }
+        // Setup spawnAttributes
+        posix_spawnattr_init(&spawnAttributes)
+        defer {
+            posix_spawnattr_destroy(&spawnAttributes)
+        }
+        var noSignals = sigset_t()
+        var allSignals = sigset_t()
+        sigemptyset(&noSignals)
+        sigfillset(&allSignals)
+        posix_spawnattr_setsigmask(&spawnAttributes, &noSignals)
+        posix_spawnattr_setsigdefault(&spawnAttributes, &allSignals)
+        // Configure spawnattr
+        var flags: Int32 = POSIX_SPAWN_CLOEXEC_DEFAULT |
+            POSIX_SPAWN_SETSIGMASK | POSIX_SPAWN_SETSIGDEF
+        if self.platformOptions.createProcessGroup {
+            flags |= POSIX_SPAWN_SETPGROUP
+        }
+        var spawnAttributeError = posix_spawnattr_setflags(&spawnAttributes, Int16(flags))
+        // Set QualityOfService
+        // spanattr_qos seems to only accept `QOS_CLASS_UTILITY` or `QOS_CLASS_BACKGROUND`
+        // and returns an error of `EINVAL` if anything else is provided
+        if spawnAttributeError == 0 && self.platformOptions.qualityOfService == .utility{
+            spawnAttributeError = posix_spawnattr_set_qos_class_np(&spawnAttributes, QOS_CLASS_UTILITY)
+        } else if spawnAttributeError == 0 && self.platformOptions.qualityOfService == .background {
+            spawnAttributeError = posix_spawnattr_set_qos_class_np(&spawnAttributes, QOS_CLASS_BACKGROUND)
+        }
+
+        // Setup cwd
+        var chdirError: Int32 = 0
+        if intendedWorkingDir != .currentWorkingDirectory {
+            chdirError = intendedWorkingDir.withPlatformString { workDir in
```

Comment:
Currently SwiftSystem's `withPlatformString` behaves differently than Foundation's `withFileRepresentation` (namely that it doesn't perform normalization/decomposition on Darwin). Do we want to use System's implementation here, or do we want to be consistent with code like `Task`/`FileManager` and use Foundation's here? (this goes for all uses of `withPlatformString`)